src/qml/components/ProxySettings.qml

```diff
@@ -33,14 +42,26 @@ ColumnLayout {
         id: defaultProxy
         Layout.fillWidth: true
         header: qsTr("IP and Port")
-        actionItem: ValueInput {
+        errorText: qsTr("Invalid IP address or port format. Please use the format '255.255.255.255:65535'.")
+        showErrorText: !loadedItem.isValidIPPort(loadedItem.description)
+        state: !defaultProxyEnable.loadedItem.checked ? "DISABLED" : "ACTIVE"
+        actionItem: IPAddressValueInput {
             parentState: defaultProxy.state
             description: "127.0.0.1:9050"
+            activeFocusOnTab: true
             onEditingFinished: {
-                defaultProxy.forceActiveFocus()
+                if (isValidIPPort(text)) {
```

Comment:
I think it might be a good idea to encapsulate the validation logic in the IPAddressValueInput component itself. I think one possible way to do that is to add a Connections element to onEditingFinished signal in the IPAddressValueInput definition so the validation can't be overridden. Then just expose a bool for if the input is valid or not.

Reply:
Ok, it makes sense, I'll try to do that. Thanks!

Reply:
Updated it with the `Connection` element and removed the `onEditingFinished` from `IPAddressValueInput`, please let me know what you think.